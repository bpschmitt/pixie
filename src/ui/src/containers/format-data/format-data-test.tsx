/*
 * Copyright 2018- The Pixie Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import * as React from 'react';
import {
  AlertData, DurationRenderer, JSONData, formatBytes, formatDuration,
} from './format-data';

describe('formatters Test', () => {
  it('should handle 0 Bytes correctly', () => {
    expect(formatBytes(0)).toEqual({ units: '\u00a0B', val: '0' });
  });

  it('should handle 0 duration correctly', () => {
    expect(formatDuration(0)).toEqual({ units: 'ns', val: '0' });
  });
  it('should handle |x| < 1  Bytes correctly', () => {
    expect(formatBytes(0.1)).toEqual({ units: '\u00a0B', val: '0.1' });
  });

  it('should handle |x| < 1 duration correctly', () => {
    expect(formatDuration(0.1)).toEqual({ units: 'ns', val: '0.1' });
  });
  it('should handle x < 0 duration correctly', () => {
    expect(formatDuration(-2)).toEqual({ units: 'ns', val: '-2' });
  });
  it('should handle x < 0 bytes correctly', () => {
    expect(formatBytes(-2048)).toEqual({ units: 'KB', val: '-2' });
  });
  it('should handle large bytes correctly', () => {
    expect(formatBytes(1024 ** 9)).toEqual({ units: 'YB', val: '1024' });
  });
  it('should handle large durations correctly', () => {
    expect(formatDuration(1000 ** 4)).toEqual({ units: '\u00A0s', val: '1000' });
  });
});

describe('DurationRenderer test', () => {
  it('should render correctly for low latency', () => {
    const component = DurationRenderer({ data: 20 * 1000 * 1000 });
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly for medium latency', () => {
    const component = DurationRenderer({ data: 250 * 1000 * 1000 });
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly for high latency', () => {
    const component = DurationRenderer({ data: 400 * 1000 * 1000 });
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });
});

describe('<AlertData/> test', () => {
  it('should render correctly for true alert', () => {
    const { container } = render(<AlertData data />);

    expect(container).toMatchSnapshot();
  });

  it('should render correctly for false alert', () => {
    const { container } = render(<AlertData data={false} />);

    expect(container).toMatchSnapshot();
  });
});

describe('<JSONData/> test', () => {
  it('should render correctly for single line', () => {
    const { container } = render(
      <JSONData
        data={{
          testString: 'a',
          testNum: 10,
          testNull: null,
          testJSON: {
            hello: 'world',
          },
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly for multiline', () => {
    const { container } = render(
      <JSONData
        data={{
          testString: 'a',
          testNum: 10,
          testNull: null,
          testJSON: {
            hello: 'world',
          },
        }}
        multiline
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render array correctly for single line', () => {
    const { container } = render(
      <JSONData
        data={['some text', 'some other text']}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render array correctly for multiline', () => {
    const { container } = render(
      <JSONData
        data={[
          { a: 1, b: { c: 'foo' } },
          { a: 3, b: null },
        ]}
        multiline
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
