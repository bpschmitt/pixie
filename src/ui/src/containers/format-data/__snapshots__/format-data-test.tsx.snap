// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertData/> test should render correctly for false alert 1`] = `
<div>
  <div
    class="AlertData-false-19"
  >
    false
  </div>
</div>
`;

exports[`<AlertData/> test should render correctly for true alert 1`] = `
<div>
  <div
    class="AlertData-true-16"
  >
    true
  </div>
</div>
`;

exports[`<JSONData/> test should render array correctly for multiline 1`] = `
<div>
  <span
    class="JSONData-base-38"
  >
    [ 
    <br />
    <span
      style="margin-left: 16px;"
    >
      <span
        class="JSONData-base-38"
      >
        { 
        <br />
        <span
          style="margin-left: 32px;"
        >
          <span
            class="JSONData-jsonKey-39"
          >
            a: 
          </span>
          <span
            class="JSONData-number-40"
          >
            <span>
              1
            </span>
          </span>
          , 
          <br />
        </span>
        <span
          style="margin-left: 32px;"
        >
          <span
            class="JSONData-jsonKey-39"
          >
            b: 
          </span>
          <span
            class="JSONData-base-38"
          >
            { 
            <br />
            <span
              style="margin-left: 48px;"
            >
              <span
                class="JSONData-jsonKey-39"
              >
                c: 
              </span>
              <span
                class="JSONData-string-42"
              >
                <span>
                  foo
                </span>
              </span>
              
              <br />
            </span>
            <span
              style="margin-left: 32px;"
            >
               }
            </span>
          </span>
          
          <br />
        </span>
        <span
          style="margin-left: 16px;"
        >
           }
        </span>
      </span>
      , 
      <br />
    </span>
    <span
      style="margin-left: 16px;"
    >
      <span
        class="JSONData-base-38"
      >
        { 
        <br />
        <span
          style="margin-left: 32px;"
        >
          <span
            class="JSONData-jsonKey-39"
          >
            a: 
          </span>
          <span
            class="JSONData-number-40"
          >
            <span>
              3
            </span>
          </span>
          , 
          <br />
        </span>
        <span
          style="margin-left: 32px;"
        >
          <span
            class="JSONData-jsonKey-39"
          >
            b: 
          </span>
          <span
            class="JSONData-null-41"
          >
            <span>
              null
            </span>
          </span>
          
          <br />
        </span>
        <span
          style="margin-left: 16px;"
        >
           }
        </span>
      </span>
      
      <br />
    </span>
    <span
      style="margin-left: 0px;"
    >
       ]
    </span>
  </span>
</div>
`;

exports[`<JSONData/> test should render array correctly for single line 1`] = `
<div>
  <span
    class="JSONData-base-32"
  >
    [ 
    <span
      style="margin-left: 0px;"
    >
      <span
        class="JSONData-string-36"
      >
        <span>
          some text
        </span>
      </span>
      , 
    </span>
    <span
      style="margin-left: 0px;"
    >
      <span
        class="JSONData-string-36"
      >
        <span>
          some other text
        </span>
      </span>
      
    </span>
    <span
      style="margin-left: 0px;"
    >
       ]
    </span>
  </span>
</div>
`;

exports[`<JSONData/> test should render correctly for multiline 1`] = `
<div>
  <span
    class="JSONData-base-26"
  >
    { 
    <br />
    <span
      style="margin-left: 16px;"
    >
      <span
        class="JSONData-jsonKey-27"
      >
        testString: 
      </span>
      <span
        class="JSONData-string-30"
      >
        <span>
          a
        </span>
      </span>
      , 
      <br />
    </span>
    <span
      style="margin-left: 16px;"
    >
      <span
        class="JSONData-jsonKey-27"
      >
        testNum: 
      </span>
      <span
        class="JSONData-number-28"
      >
        <span>
          10
        </span>
      </span>
      , 
      <br />
    </span>
    <span
      style="margin-left: 16px;"
    >
      <span
        class="JSONData-jsonKey-27"
      >
        testNull: 
      </span>
      <span
        class="JSONData-null-29"
      >
        <span>
          null
        </span>
      </span>
      , 
      <br />
    </span>
    <span
      style="margin-left: 16px;"
    >
      <span
        class="JSONData-jsonKey-27"
      >
        testJSON: 
      </span>
      <span
        class="JSONData-base-26"
      >
        { 
        <br />
        <span
          style="margin-left: 32px;"
        >
          <span
            class="JSONData-jsonKey-27"
          >
            hello: 
          </span>
          <span
            class="JSONData-string-30"
          >
            <span>
              world
            </span>
          </span>
          
          <br />
        </span>
        <span
          style="margin-left: 16px;"
        >
           }
        </span>
      </span>
      
      <br />
    </span>
    <span
      style="margin-left: 0px;"
    >
       }
    </span>
  </span>
</div>
`;

exports[`<JSONData/> test should render correctly for single line 1`] = `
<div>
  <span
    class="JSONData-base-20"
  >
    { 
    <span
      style="margin-left: 0px;"
    >
      <span
        class="JSONData-jsonKey-21"
      >
        testString: 
      </span>
      <span
        class="JSONData-string-24"
      >
        <span>
          a
        </span>
      </span>
      , 
    </span>
    <span
      style="margin-left: 0px;"
    >
      <span
        class="JSONData-jsonKey-21"
      >
        testNum: 
      </span>
      <span
        class="JSONData-number-22"
      >
        <span>
          10
        </span>
      </span>
      , 
    </span>
    <span
      style="margin-left: 0px;"
    >
      <span
        class="JSONData-jsonKey-21"
      >
        testNull: 
      </span>
      <span
        class="JSONData-null-23"
      >
        <span>
          null
        </span>
      </span>
      , 
    </span>
    <span
      style="margin-left: 0px;"
    >
      <span
        class="JSONData-jsonKey-21"
      >
        testJSON: 
      </span>
      <span
        class="JSONData-base-20"
      >
        { 
        <span
          style="margin-left: 0px;"
        >
          <span
            class="JSONData-jsonKey-21"
          >
            hello: 
          </span>
          <span
            class="JSONData-string-24"
          >
            <span>
              world
            </span>
          </span>
          
        </span>
        <span
          style="margin-left: 0px;"
        >
           }
        </span>
      </span>
      
    </span>
    <span
      style="margin-left: 0px;"
    >
       }
    </span>
  </span>
</div>
`;

exports[`DurationRenderer test should render correctly for high latency 1`] = `
<div>
  <div
    class="GaugeData-high-13"
  >
    <span
      class="RenderValueWithUnits-value-15"
    >
      400 
    </span>
    <span
      class="RenderValueWithUnits-units-14"
    >
      ms
    </span>
  </div>
</div>
`;

exports[`DurationRenderer test should render correctly for low latency 1`] = `
<div>
  <div
    class="GaugeData-low-1"
  >
    <span
      class="RenderValueWithUnits-value-5"
    >
      20 
    </span>
    <span
      class="RenderValueWithUnits-units-4"
    >
      ms
    </span>
  </div>
</div>
`;

exports[`DurationRenderer test should render correctly for medium latency 1`] = `
<div>
  <div
    class="GaugeData-med-7"
  >
    <span
      class="RenderValueWithUnits-value-10"
    >
      250 
    </span>
    <span
      class="RenderValueWithUnits-units-9"
    >
      ms
    </span>
  </div>
</div>
`;
